manual module listing

George-Elhamy opened the pull request:
<!--- Provide a general short and descriptive title above -->

## Description
<!--- Describe your changes in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

## Screenshots (only if appropriate, delete if not):

## Types of changes
<!--- What types of changes does your code introduce? -->
<!--- Put an 'x' in all the boxes that apply, e.g. "[x]". -->
<!--- You can also create this PR and afterwards check the boxes by clicking on them. -->
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change (Causes existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an 'x' in all the boxes that apply, e.g. "[x]". -->
<!--- If you're unsure about any of these, ask. -->
<!--- You can also create this PR and afterwards check the boxes by clicking on them. -->
- [ ] My code follows the code style of this project.
- [ ] I have documented my changes in `/docs/changelog.rst`.
- [ ] My change requires additional/updated documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added/updated the config example for any module docstrings as necessary.
- [ ] I have checked that the change does not contain obvious errors
(syntax, indentation, mutable default values, etc.).
- [ ] I have tested my changes using 'Load all modules' on the default dummy configuration.
- [ ] All changed Jupyter notebooks have been stripped of their output cells.

George-Elhamy committed:
manual module listing

prithviulm commented on calculate_documentation_coverage.py:
```diff
@@ -44,7 +44,7 @@ def parse_files(directory):
 
 # Specify the directory containing your Python files
-directory = "../src/qudi/"
+directory = "src/qudi"
```
Would suggest using os.path.join since different operating systems use different slash directions (forward vs backward) to specify directories.

George-Elhamy replied:
I already change it, so have a look on it and approve if it looks good!

prithviulm reviewed (changes requested):
Looks good! Just a minor comment about using `os.path.join` where possible.

George-Elhamy committed:
fixing path

George-Elhamy committed:
Fixing Documentations

George-Elhamy committed:
Fixing top bar and toctrees

prithviulm commented on calculate_documentation_coverage.py:
```diff
@@ -44,7 +44,8 @@ def parse_files(directory):
 
 # Specify the directory containing your Python files
-directory = "../src/qudi/"
+directory = os.path.join("src", "qudi")
```
I don't think this is equivalent to what it was before since it's missing the `../`. Could you check this?

George-Elhamy replied:
Yes, I know. But it actually turned out to be that the one without the ../ is the correct one so that's why i made it like this.

prithviulm commented:
Also I see there are hundreds of files now added for the api autodoc in an `_autosummary` folder. Be careful, these should be generated by the docs building process (correct me if I'm wrong) and so should not be part of the repo. Please update the `.gitignore` (there is an existing pattern there to ignore `_autosummary` already but one directory higher) to handle this and remove these files.

George-Elhamy committed:
Few syntax errors

George-Elhamy committed:
removing docs/api_reference/_autosummary/ from tracking

George-Elhamy commented:
> Also I see there are hundreds of files now added for the api autodoc in an `_autosummary` folder. Be careful, these should be generated by the docs building process (correct me if I'm wrong) and so should not be part of the repo. Please update the `.gitignore` (there is an existing pattern there to ignore `_autosummary` already but one directory higher) to handle this and remove these files.

I've already added the `_autosummary` folder to the `.gitignore` and stopped its tracking in the branch, so check it and let me know if there is anything else to modify.

George-Elhamy committed:
modifying the template

George-Elhamy committed:
logo

George-Elhamy committed:
theme fixing

prithviulm commented on calculate_documentation_coverage.py:
Would be good to have docstrings and be pep8 compliant (e.g. use of snake case) but it's fine

George-Elhamy committed:
heirarchy modification

George-Elhamy committed:
Fixing hyperlinks

George-Elhamy committed:
hyperlinks 2.0

George-Elhamy committed:
hyperlinks cntd

George-Elhamy committed:
Rendering modifications

George-Elhamy committed:
Jupyter

George-Elhamy committed:
data fitting

George-Elhamy committed:
fix qudi iqo_modules

George-Elhamy committed:
landing page and some bugs

George-Elhamy committed:
update toml

George-Elhamy committed:
notes
